tests/components/media_player/test_directv.py

```diff
@@ -0,0 +1,507 @@
+"""The tests for the DirecTV Media player platform."""
+from unittest.mock import call, patch
+
+from datetime import datetime, timedelta
+import pytest
+
+import homeassistant.components.media_player as mp
+from homeassistant.components.media_player import (
+    ATTR_MEDIA_CONTENT_ID, ATTR_MEDIA_CONTENT_TYPE, ATTR_MEDIA_ENQUEUE, DOMAIN,
+    SERVICE_PLAY_MEDIA)
+from homeassistant.components.media_player.directv import (
+    ATTR_MEDIA_CURRENTLY_RECORDING, ATTR_MEDIA_RATING, ATTR_MEDIA_RECORDED,
+    ATTR_MEDIA_START_TIME, DATA_DIRECTV, DEFAULT_DEVICE, DEFAULT_PORT)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, CONF_DEVICE, CONF_HOST, CONF_NAME, CONF_PORT,
+    SERVICE_MEDIA_NEXT_TRACK, SERVICE_MEDIA_PAUSE, SERVICE_MEDIA_PLAY,
+    SERVICE_MEDIA_PREVIOUS_TRACK, SERVICE_MEDIA_STOP, SERVICE_TURN_OFF,
+    SERVICE_TURN_ON, STATE_OFF, STATE_PAUSED, STATE_PLAYING, STATE_UNAVAILABLE)
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.setup import async_setup_component
+import homeassistant.util.dt as dt_util
+
+from tests.common import MockDependency
+
+CLIENT_ENTITY_ID = 'media_player.client_dvr'
+MAIN_ENTITY_ID = 'media_player.main_dvr'
+IP_ADDRESS = '127.0.0.1'
+
+DISCOVERY_INFO = {
+    'host': IP_ADDRESS,
+    'serial': 1234
+}
+
+LIVE = {
+    "callsign": "HASSTV",
+    "date": "20181110",
+    "duration": 3600,
+    "isOffAir": False,
+    "isPclocked": 1,
+    "isPpv": False,
+    "isRecording": False,
+    "isVod": False,
+    "major": 202,
+    "minor": 65535,
+    "offset": 1,
+    "programId": "102454523",
+    "rating": "No Rating",
+    "startTime": 1541876400,
+    "stationId": 3900947,
+    "title": "Using Home Assistant to automate your home"
+}
+
+LOCATIONS = [
+    {
+        'locationName': 'Main DVR',
+        'clientAddr': DEFAULT_DEVICE
+    }
+]
+
+RECORDING = {
+    "callsign": "HASSTV",
+    "date": "20181110",
+    "duration": 3600,
+    "isOffAir": False,
+    "isPclocked": 1,
+    "isPpv": False,
+    "isRecording": True,
+    "isVod": False,
+    "major": 202,
+    "minor": 65535,
+    "offset": 1,
+    "programId": "102454523",
+    "rating": "No Rating",
+    "startTime": 1541876400,
+    "stationId": 3900947,
+    "title": "Using Home Assistant to automate your home",
+    'uniqueId': '12345',
+    'episodeTitle': 'Configure DirecTV platform.'
+}
+
+WORKING_CONFIG = {
+    'media_player': {
+        'platform': 'directv',
+        CONF_HOST: IP_ADDRESS,
+        CONF_NAME: 'Main DVR',
+        CONF_PORT: DEFAULT_PORT,
+        CONF_DEVICE: DEFAULT_DEVICE
+    }
+}
+
+
+@pytest.fixture
+async def platforms(hass):
```

Comment:
Standard pytest fixtures can't be coroutines. If we need to schedule a coroutine in a fixture, use `loop.run_until_complete`.

Reply:
If I use:
```
setup_component(hass, mp.DOMAIN, config)
hass.block_till_done()
```
to setup the component within the fixture then it hangs.

If I use:
```
await async_setup_component(hass, mp.DOMAIN, config)
await hass.async_block_till_done()
```
then it works but only if I use:
`async def platforms(hass)`

What is the right method then to setup a component within a fixture?

Reply:
```py
hass.loop.run_until_complete(
    async_setup_component(...))
```

Reply:
Think I figured it out. For some reason I always seem to get to the answer right after I ask here. Next time just going to ask before I start looking and wasting time, then once asked the answer will appear on my screen right away. :-)